library/psa_crypto.c

```diff
@@ -6489,32 +6489,12 @@ static psa_status_t psa_key_derivation_input_integer_internal(
     psa_status_t status;
     psa_algorithm_t kdf_alg = psa_key_derivation_get_kdf_alg(operation);
 
-#if defined(BUILTIN_ALG_ANY_HKDF)
-    if (PSA_ALG_IS_ANY_HKDF(kdf_alg)) {
-        status = PSA_ERROR_INVALID_ARGUMENT;
-    } else
-#endif /* BUILTIN_ALG_ANY_HKDF */
-#if defined(MBEDTLS_PSA_BUILTIN_ALG_TLS12_PRF)
-    if (PSA_ALG_IS_TLS12_PRF(kdf_alg)) {
-        status = PSA_ERROR_INVALID_ARGUMENT;
-    } else
-#endif /* MBEDTLS_PSA_BUILTIN_ALG_TLS12_PRF */
-#if defined(MBEDTLS_PSA_BUILTIN_ALG_TLS12_PSK_TO_MS)
-    if (PSA_ALG_IS_TLS12_PSK_TO_MS(kdf_alg)) {
-        status = PSA_ERROR_INVALID_ARGUMENT;
-    } else
-#endif /* MBEDTLS_PSA_BUILTIN_ALG_TLS12_PSK_TO_MS */
-#if defined(MBEDTLS_PSA_BUILTIN_ALG_TLS12_ECJPAKE_TO_PMS)
-    if (kdf_alg == PSA_ALG_TLS12_ECJPAKE_TO_PMS) {
-        status = PSA_ERROR_INVALID_ARGUMENT;
-    } else
-#endif /* MBEDTLS_PSA_BUILTIN_ALG_TLS12_ECJPAKE_TO_PMS */
```

Comment:
Note on commit structure: we want one commit per logical change, with a meaningful description. Here we should have had one commit for this, then another one for the test changes. No need to rewrite the history now that it's done, just something to keep in mind for the future.

Reply:
Btw, we've published some documentation in the meantime: https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/